blueprints/test-framework-detector.js

```diff
@@ -12,7 +14,12 @@ module.exports = function(blueprint) {
     if ('ember-cli-qunit' in dependencies) {
       type = 'qunit';
     } else if ('ember-cli-mocha' in dependencies) {
-      type = 'mocha';
+      if (fs.existsSync(this.path + '/mocha-0.12-files') &&
+        semver.satisfies('0.12.0', dependencies['ember-cli-mocha'])) {
+        type = 'mocha-0.12';
+      } else {
+        type = 'mocha';
```

Comment:
@rwjblue that is precisely what I meant with "small issue" above. I've fixed it in https://github.com/emberjs/data/pull/4691 by using the `VersionChecker` class and generating fake package manifests for the mocha blueprint tests. will update this PR now too.